crates/typst-cli/src/compile.rs

```diff
@@ -95,7 +95,7 @@ pub fn compile_once(
             print_diagnostics(world, &[], &warnings, command.common.diagnostic_format)
                 .map_err(|_| "failed to print diagnostics")?;
 
-            if let Some(open) = command.open.take() {
+            if let Some(open) = command.open.clone().take() {
```

Comment:
This is problematic because we call `take()` specifically for the side effect. Otherwise, the file will open over and over again. Especially for typst watch, we must retain that state across multiple compile_once calls.

Reply:
Ah, I had missed that. I pushed a simple fix that just clones the command before padding it to watch or compile so that they can modify their copy of it.